tests/test_install.py

```diff
@@ -32,7 +32,8 @@
 pytestmark = pytest.mark.slow
 
 
-untar = shell.ShellCommand(("gtar" if command_exists("gtar") else "tar") + " xvzkf")
+tar = shell.get_executable("gtar" if command_exists("gtar") else "tar")
```

Comment:
Judging by the results of the CI in the last results: it seems that when `shell=False`, Python has problems to find the correct executable.

It looks like the env variables are available for the children process after it starts, but `PATH` is not being used to find the correct executable when a full path is not provided... (the tests just started working when `shell.get_executable` got introduced here)

Reply:
I did a little experiment to understand more about this limitation:

On Linux:
```bash
echo -n '#!/bin/sh\necho $MYVAR' > test/tar
chmod +x test/tar
PATH="./test:$PATH" MYVAR=42 python3 -c 'import subprocess; subprocess.run(["tar", "-h"])'
# => 42
```
∴ on Linux, `subprocess.run` is able to find the correct executable considering the current value of `PATH`.

On Windows, I created the files `other.bat` and `tar.bat` inside the `test` folder in my current working dirctory, both containing the following code:
```batch
@echo off
echo %MYVAR%
```
Then I run (on CMD):
```batch
set PATH=test;%PATH%
set MYVAR=42
other
:: => 42
tar
:: => 42
python -c "import os; print(os.getenv('PATH'))"
:: => test;C:\Python38\Scripts\;C:\Python38\; ... ;C:\WINDOWS\system32; ...
python -c "import subprocess; subprocess.run(['other', '-h'])"
:: => ... FileNotFoundError: [WinError 2] The system cannot find the file specified ...
python -c "import subprocess; subprocess.run(['tar', '-h'])"
:: => tar(bsdtar): manipulate archive files ...
python -c "import subprocess; subprocess.run(['tar.bat', '-h'])"
:: => 42
copy C:\WINDOWS\system32\tar.exe .\test\foobar.exe
python -c "import subprocess; subprocess.run(['foobar', '-h'])"
:: => foobar(bsdtar): manipulate archive files ...
rename rename test\foobar.exe notepad.exe
python -c "import subprocess; subprocess.run(['notepad', '-h'])"
:: Opens a notepad window
python -c "import os, subprocess; print(os.getenv('PATH'));  subprocess.run(['notepad', '-h'])"
:: Prints "test;C:\Python38\Scripts\;C:\Python38\; ... ;C:\WINDOWS\system32; ..." but still opens a notepad window
```
∴ on Windows, `subprocess.run` will only find the correct file considering `PATH` if the file extension is given explicitly as part of the file name (e.g. `.bat`), or when it is `.exe` **BUT** executables on `C:\WINDOWS\system32` will have priority, even if the `PATH` is adjusted in a way that `C:\WINDOWS\system32` comes last.

In the case of the failing tests, what was happening is that `C:\WINDOWS\system32\tar.exe` was given priority over `C:\tools\GitForWindows.2.27.0\tools\usr\\bin\tar.exe`